packages/scm/src/browser/scm-frontend-module.ts

```diff
@@ -74,4 +101,22 @@ export default new ContainerModule(bind => {
     bind(ScmAvatarService).toSelf().inSingletonScope();
 
     bindDirtyDiff(bind);
+
+    bind(ScmTreeLabelProvider).toSelf().inSingletonScope();
+    bind(LabelProviderContribution).toService(ScmTreeLabelProvider);
 });
+
+export function createFileChangeTreeWidget(parent: interfaces.Container): ScmTreeWidget {
```

Comment:
`createScmTreeWidget`

It's usually better to have a function which returns a container, so extenders can call this function rebind services and then get a widget. Otherwise they have to copy the entire function.

Reply:
now changed to return container.